app/views/messages/_message.html.erb

```diff
@@ -13,11 +13,31 @@ end %>
 <% scroll_down = scroll_down || initial_page_load %>
 <% thinking ||= last_message && message.content_text == "" && message.not_cancelled? %>
 <% done_thinking = !thinking && message.content_text.present? %>
-<% debug_tool_messages = false %>
+<% memory_updated = from_name_for(message) == "Tool" && message.content_tool_calls.dig(:function, :name) == "memory_remember_detail_about_user" %>
+
+<% if memory_updated %>
+  <div class="flex mb-5 ml-9 sm:ml-3 md:ml-8">
+    <%= icon "archive-box",
+      variant: :outline,
+      size: 22,
+      class: %|
+        ml-1
+        mr-1
+        |
+    %>
+    <div class="ml-1">
+      <%= format_for_display(message.content_text) %>
+    </div>
+    <%= link_to "(manage memories)\n", settings_memories_path,
+      data: { turbo_frame: "_top" },
+      class: "ml-1 text-gray-600 dark:text-gray-300"
+    %>
+  </div>
+<% end %>
 
 <div
   id="<%= dom_id message %>"
-  class="mb-5 ml-9 sm:ml-3 md:ml-8 group <%= message.only_tool_response? && !debug_tool_messages && "hidden" %>"
+  class="mb-5 ml-9 sm:ml-3 md:ml-8 group <%= message.only_tool_response? && "hidden" %>"
```

Comment:
reverting the debug and using your helper, this would become:

```suggestion
  class="mb-5 ml-9 sm:ml-3 md:ml-8 group <%= message.only_tool_response? && !memory_updated?(message) && !debug_tool_message && "hidden" %>"
```